src/lib/data/customer.ts

```diff
@@ -0,0 +1,113 @@
+"use server"
+
+import { sdk } from "@lib/config"
+import medusaError from "@lib/util/medusa-error"
+import { HttpTypes } from "@medusajs/types"
+import { revalidateTag } from "next/cache"
+import { cookies } from "next/headers"
+import { redirect } from "next/navigation"
+import { cache } from "react"
+
+// Note: For some weird reason the response is SOMETIMES a promise here... does it get called like a server action? Investigate.
```

Comment:
@VariableVic what I would have expected is that, since all the methods that call `getAuthHeaders` are server actions, this method should just be a normal function call on the server. I'll try your suggestion and see if that fixes it.